### ISSUE TITLE
Provide a Flatpak package

### ISSUE THREAD

**Issue by Alexander-Wilms** [ValveSoftware/steam-for-linux#4473]
Citing [flatpak.org](flatpak.org):

"Flatpak is the new framework for desktop applications on Linux
Distributing applications on Linux is a pain: different distributions in multiple versions, each with their own versions of libraries and packaging formats. Flatpak is here to change all that. It allows the same app to be installed on different Linux distributions, including different versions. And it has been designed from the ground up with security in mind, so that apps are isolated from each other and from the host system."

By providing a single Flatpak (previously known as xdg-app) package for all distros it should be easier to reach a wider audience and avoid library conflicts like [https://github.com/ValveSoftware/steam-for-linux/issues/3273](https://github.com/ValveSoftware/steam-for-linux/issues/3273).

**Comment by ekaris** [comment]
This would be great, please do this!

**Comment by mg-13** [comment]
Why not snap?
Canonical standardised this package format for the same use case, when apps depend on specific versions of libs. snapd has been ported to other distributions already: [ubuntu insights](https://insights.ubuntu.com/2016/06/14/universal-snap-packages-launch-on-multiple-linux-distros/)

**Comment by ekaris** [comment]
Because flatpak is better

**Comment by borring** [comment]
@mg-13 flatpak also works with multiple distros. And with flatpak, Valve will be able to host their own repository.

For Steam in particular, I don't think it matters which standard they choose. In general though, I think Flatpak is the better solution.

**Comment by mrkara** [comment]
This would relieve the pain of installing steam on different linux distros. I hope it is done in the near future.

**Comment by tapir** [comment]
Current steam distribution is for the lack of a better word: Disaster. Please, please, please provide a flatpak and an official valve repo to host it so we can get updates.

**Comment by borring** [comment]
@tapir Something like this?

	Valve repository:
	com.valvesoftware.Platform
	com.valvesoftware.Sdk
	com.valvesoftware.Steam

	com.blenderfoundation.Blender
	com.valvesoftware.CounterStrike_GO

Valve distributing via flatpak would be awesome. The problem is that flatpak doesn't (yet?) support MacOS. I imagine that would be a pretty important factor.

**Comment by aelog** [comment]
Steam is exactly what flatpak was designed for (see the concept of flatpak runtimes). Please make this happen.

**Comment by baconicsynergy** [comment]
Please let this happen. It would be simply incredible

**Comment by Alexander-Wilms** [comment]
In case the Steam binary may not be redistributed, the flatpak could be created like this: https://blogs.gnome.org/alexl/2016/10/26/distributing-spotify-as-a-flatpak/

This could probably even be done by the community.

**Comment by CreamyCookie** [comment]
@ekaris Why is flatpak better? Aren't they pretty similar? (For example, they can both work on multiple distros)

**Comment by tapir** [comment]
Because every distro except Ubuntu will probably adopt flatpak. Eventhough
Snap now works cross-distro, Canonical is not be trusted because there is
countless examples that show that they don't care about community.
Canonical is an island in the sea of free software.

**Comment by AsciiWolf** [comment]
@tapir As far as I know, _flatpak_ and _gnome-software-plugin-flatpak_ packages are already available in Ubuntu 17.04, just not installed by default.

**Comment by nickik** [comment]
This would be great, Steam constantly has problems on Arch Linux.

**Comment by ekaris** [comment]
Flatpak is the one to choose.

https://kamikazow.wordpress.com/2017/02/09/adoption-of-flatpak-vs-snap/

**Comment by troyfolger** [comment]
another testimonial: https://solus-project.com/2017/01/18/adopting-flatpak-to-reassemble-third-party-applications/

**Comment by carlosnewmusic** [comment]
You want to stop arguing in which encapsulation is better if snap flatpak or appimage (although the latter installs / runs on any system) do the package and already because I take the life looking for solution to dependency problems in debian ...

**Comment by Vistaus** [comment]
@borring Steam currently supplies Steam officially only as a DEB package. macOS doesn't support DEB packages. So I doubt it'd be a big factor regarding Flatpak.

**Comment by Vistaus** [comment]
@tapir Currently, Ubuntu supports Flatpak as it ships with GNOME Software by default (which can handle Flatpak packages just fine). But I don't know if they will keep it like that, they'll most likely patch it to only handle Snap packages in the future.

**Comment by Vistaus** [comment]
Yes, please let this happen! This way I can install Steam on my MicroSD rather than on my SSD (which is rather small and Solus and Vivaldi and a few other apps already take up quite some space).

**Comment by Alexander-Wilms** [comment]
Someone from Endless has been working on flatpaking Steam: https://github.com/endlessm/steam-flatpak

I haven't tried building it yet.

**Comment by probonopd** [comment]
@Vistaus 
> AppImage is a pretty nice piece of software, but it doesn't really incorporate any modern features like sandboxing and stuff.

Not true, Firejail has native AppImage support. And the Bubblewrap sandbox used by Flatpak can be used to run AppImages too.

> That really is needed these days.

Really? Don't you trust Steam? Then better not run it in the first place...

> and rewriting AppImage to incoporate such features is useless because it would end up more or less the same as Flatpak / Snap anyway

Not at all. The core principle of AppImage is "one app = one file". AppImage differs from other solutions in that:

- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary. The uninstalled version is the same as the installed version. With other systems, it is difficult to get the software from machine A to machine B once it has been installed on machine A. With AppImage, you simply copy over one file, done
- No root needed
- No system libraries changed
- Works out of the box, no installation of runtimes needed (unlike Snap and Flatpak)
- Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Works on Live ISOs (unlike Flatpak)
- Can use the same AppImages when dual-booting multiple distributions. With other solutions, the software is intermingled with the system in a way that makes it hard to use in multi-boot configurations

**Comment by Vistaus** [comment]
@probonopd 

"Not true, Firejail has native AppImage support. And the Bubblewrap sandbox used by Flatpak can be used to run AppImages too."

Oh, so Firejail and Bubblewrap are integrated into AppImage? I wasn't aware of that! [/s]

Dude, come on. You didn't even read my post correctly. I said that AppImage doesn't have sandboxing integrated, I never said you can't use 3rd party apps to sandbox it.

"Really? Don't you trust Steam? Then better not run it in the first place..."

I do trust Steam, but who says other AppImage-packaged apps are to be trusted?

**Comment by probonopd** [comment]
@Vistaus the beauty of AppImage is that it lets you choose which sandbox (or none) you like. The optional `appimaged` daemon can make use of Firejail if it is present. You should decide on a per-application basis what (not) to trust, there is no general rule. This goes for any kind of application, regardless of the package format.

**Comment by Vistaus** [comment]
@Alexander-Wilms Thanks for sharing! I just tried it but it it's broken atm:

[11:17:08] vistaus@vistaus-solus /media/MicroSD/Apps/steam-flatpak (2) 
> make -j3
ostree init --mode=archive-z2 --repo=repo
make: ostree: Command not found
make: *** [Makefile:10: repo] Error 127
[11:17:09] vistaus@vistaus-solus /media/MicroSD/Apps/steam-flatpak (2) 
> 

Even though ostree and ostree-devel are installed, so...

**Comment by Vistaus** [comment]
@probonopd Sure, that might be true but that's still different from what I said. I said it doesn't have sandboxing integrated, to which you responded "not true". But it is true, 'cause Firejail isn't integrated.

Also, setting up Firejail is not something everyone can just do. I have no idea how to set it up, let alone someone who's new to Linux and just downloaded his/her first AppImage...

**Comment by probonopd** [comment]
Come on, Flatpak also needs to be set up (at least on Ubuntu).

**Comment by Vistaus** [comment]
@probonopd Flatpak is set up by default on Ubuntu GNOME. On Ubuntu Unity, I've heard that it's not installed by default but it's just a matter of installing the flatpak package. No need to set up rules and stuff like you have to do in Firejail.
(I haven't used Ubuntu in a long time, so maybe there's more to it but this is what I've heard from people on Phoronix)

**Comment by probonopd** [comment]
> Flatpak is set up by default on Ubuntu GNOME. 

__Not true__:  http://cdimage.ubuntu.com/ubuntu-gnome/releases/16.10/release/ubuntu-gnome-16.10-desktop-amd64.manifest

People should __not__ assume Flatpak to be preinstalled everywhere because it is not going to happen. That is my point - why bother with a technology that needs to have a runtime to be installed on the system first, when you can have something that already works on all major distributions today.

**Comment by Vistaus** [comment]
@probonopd Yes true: http://cdimage.ubuntu.com/ubuntu-gnome/releases/17.04/beta-2/ubuntu-gnome-17.04-beta2-desktop-amd64.manifest

**Comment by probonopd** [comment]
OK, in a beta. But not in the release, let alone in LTS.

**Comment by Vistaus** [comment]
@probonopd One of the 17.04 Ubuntu GNOME developers has confirmed somewhere on Phoronix that it will be default on 17.04 final as well.

**Comment by probonopd** [comment]
But then, who uses Ubuntu GNOME. Most people use plain Ubuntu.

**Comment by Vistaus** [comment]
@probonopd Judging by Phoronix readers, more people have started using it since 16.10 But even then, installing Flatpak on Ubuntu Unity is still much easier than setting up Firejail. But yes, if you don't care about sandboxing, then AppImage is much easier.

**Comment by probonopd** [comment]
As expected, does not work on ubuntu-gnome-17.04-beta2-desktop-amd64.iso:

error: While pulling runtime/org.freedesktop.Platform/x86_64/1.4 from remote gnome: Unable to set xattr: Operation not supported

**Comment by AsciiWolf** [comment]
@probonopd This issue looks to be present only when using a live iso, as noted [here](https://github.com/flatpak/flatpak/issues/126#issuecomment-230360000).

**Comment by Vistaus** [comment]
I'm not sure if Ubuntu works differently or if it's bug. Try the 17.04 daily ISO from yesterday to see if you can reproduce it or try installing and see if it works after installing (it should at least work after installing, I'm told).

**Comment by probonopd** [comment]
I don't want to _install_ it, I want to _download and run_ it without it needing to be installed...

**Comment by AsciiWolf** [comment]
Ubuntu is [switching back to GNOME](https://phoronix.com/scan.php?page=news_item&px=Ubuntu-Dropping-Unity). It probably means that Flatpak (along with desktop portals and gnome-software-flatpak plugin) will be installed by-default.

**Comment by probonopd** [comment]
Let's wait and see, but there are many other desktop environments besides GNOME too.

**Comment by troyfolger** [comment]
I'm enjoying the discussion, but mostly I think that ANY migration to a container approach that encapsulates run-time dependencies would be welcome. I have spent more time trying to get Steam working than any other package (in the last 4 years or so). I think if just ONE of the varying approaches gets sorted out, it will be trivially easy to move to another approach, and eventually 3rd-party packagers will pick up the packages and we can all choose the one that works best for our environment.

In other words ... pick one and get it working, and it is all downhill from there. Um, please.  :)

**Comment by probonopd** [comment]
I looked into making an AppImage out of the Steam client, but it's a mess... not the AppImage part, but the Steam client. I have documented it here:

https://github.com/probonopd/AppImageKit/wiki/Bundling-Steam

* There is a `deb`, then there is the same thing as a  `.tar.gz` but even if you download that, it insists on installing a lot of i386 libraries into your system using `apt`. I thought there was a Steam Runtime for this reason, so that it would __not__ have to mess with the base system?
* After having done that, it tries to launch something which ends in a `libGL error`
* If you run another binary randomly placed into `/home/me/.steam/bin/steam` then it starts to download a whole lot of other stuff, further cluttering `$HOME`
* After having done that, it tries to launch something which ends in a `Failed to load steamui.so - dlerror(): steamui.so`
* If you run another binary randomly placed into `/home/me/.steam/steam/steam.sh`  then it finally launches
* If you want to launch this directly, you have to `LD_LIBRARY_PATH=/home/me/.steam/steam/ubuntu12_32:/home/me/.steam/steam/ubuntu12_32/panorama:/lib/i386-linux-gnu:/usr/lib/i386-linux-gnu:/usr/lib/i386-linux-gnu/mesa:/usr/local/lib:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu/mesa-egl:/usr/lib/x86_64-linux-gnu/mesa:/lib:/usr/lib:/home/me/.steam/steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu:/home/me/.steam/steam/ubuntu12_32/steam-runtime/i386/lib:/home/me/.steam/steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu:/home/me/.steam/steam/ubuntu12_32/steam-runtime/i386/usr/lib:/home/me/.steam/steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu:/home/me/.steam/steam/ubuntu12_32/steam-runtime/amd64/lib:/home/me/.steam/steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu:/home/me/.steam/steam/ubuntu12_32/steam-runtime/amd64/usr/lib: /home/me/.steam/steam/ubuntu12_32/steam`
* Now, why is it using stuff from e.g., `/usr/lib/i386-linux-gnu/libstdc++.so.6` and not from its own runtime it comes bundled with? It would be available in `/home/me/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libstdc++.so.6`
* Hey, we can `patchelf --set-rpath '$ORIGIN:$ORIGIN/steam-runtime/i386/usr/lib/i386-linux-gnu/' /home/me/.steam/steam/ubuntu12_32/steam`
* At this point, Steam decides that it wants to reinstall itself

This is where I gave up. I am pretty confident I could make an AppImage that doesn't suck and would not have any dependencies on the base systems other than a Kernel and FUSE, but for this to happen the Steam guys would need to provide something sane, i.e., something that can be installed to a read-only location, that doesn't insist on cluttering `$HOME`, and wouldn't reinstall itself.

**Comment by troyfolger** [comment]
Yeah, exactly, probonopd. I think the kindest way to put it is that there
are a number of distro-specific assumptions that are part of the Steam
client and it is probably due a redesign. I appreciate the fact that Steam
is doing a lot, it is providing a 'platform' for games to run on that
attempts to minimize extra development for the game developer. But
precisely for the errors you list in your message, a container format is
just what the Steam dev team should target. Get it working reliable and
cleanly in a container, provide access to host graphics, provide a sane
means for game storage (the 'library'), provide a sane means for user data
(preferences, saved games, etc) ... it isn't a 'snap' (pun intended),
perhaps, but it would solve lots of problems in the longer term, and could
improve Steam's ability to deploy to more platforms.

I think the biggest hurdle would be reliable access to host graphic
capabilities. Not impossible, but the part that would require the most
testing/tweaking.

**Comment by neagix** [comment]
>  I appreciate the fact that Steam
is doing a lot, it is providing a 'platform' for games to run on that
attempts to minimize extra development for the game developer.

Well, I do **not** for one. If you advertise "for Linux" when customers are buying games, and you provide an "official" Linux client, then you should put some serious development effort in it rather than "spare time" development. I understand the Linux profit is not as good as the other platforms, but either you do it good or don't try at all.

The situation of the Linux client is simply ridiculous at the moment; you can bet that if I had known it would end up like this, I would have bought exactly **0** of the games I bought because I thought I could play them on my Linux devices.

Sorry if this looks like a rant, but long story short if you sell games for Linux then understand the Linux market and provide proper software. There are plenty of successful commercial examples nowadays, it's no more 2001.

**Comment by elvisangelaccio** [comment]
http://www.phoronix.com/scan.php?page=news_item&px=Endless-Flatpaks-Steam

**Comment by tidux** [comment]
@elvisangelaccio they seem to be doing everything in their power to obfuscate where the Flatpak repos actually are, at least when looking at their website.

**Comment by Alexander-Wilms** [comment]
Steam is now also available from flathub.org: https://github.com/flathub/com.valvesoftware.Steam

**Comment by haevalencia** [comment]
@Alexander-Wilms it's official?

**Comment by Alexander-Wilms** [comment]
@haevalencia No, that manifest was written by Flatpak devs. However, it's Flathub's policy to give the application developers write-access if they request it. So in theory, it could become official.

**Comment by probonopd** [comment]
Does it somehow disable the annoying online-updater? This is totally destroying the experience of AppImages, Flatpaks, Snaps, etc.

**Comment by Alexander-Wilms** [comment]
@probonopd No, it doesn't.

**Comment by probonopd** [comment]
> Ubuntu is switching back to GNOME. It probably means that Flatpak (along with desktop portals and gnome-software-flatpak plugin) will be installed by-default.

So, let's see, this is on `ubuntu-17.10-desktop-amd64.iso` out-of-the-box:

![](https://user-images.githubusercontent.com/2480569/32698333-25d85a20-c7a4-11e7-9de7-6926c27fd6cb.png)

**Comment by Alexander-Wilms** [comment]
I get the hint  "You will have to enable the component called 'universe'".

`alexander@qemu:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="17.10 (Artful Aardvark)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 17.10"
VERSION_ID="17.10"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=artful
UBUNTU_CODENAME=artful
alexander@qemu:~$ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
The program 'flatpak' is currently not installed. You can install it by typing:
sudo apt install flatpak
You will have to enable the component called 'universe'
alexander@qemu:~$ sudo apt-add-repository universe
'universe' distribution component enabled for all sources.
alexander@qemu:~$ sudo apt-get update
Get:1 http://security.ubuntu.com/ubuntu artful-security InRelease [76,7 kB]
Hit:2 http://de.archive.ubuntu.com/ubuntu artful InRelease
Hit:3 http://de.archive.ubuntu.com/ubuntu artful-updates InRelease                  
Hit:4 http://de.archive.ubuntu.com/ubuntu artful-backports InRelease                 
Get:5 http://de.archive.ubuntu.com/ubuntu artful/universe i386 Packages [8.066 kB]
Get:6 http://security.ubuntu.com/ubuntu artful-security/universe i386 Packages [5.812 B]
Get:7 http://security.ubuntu.com/ubuntu artful-security/universe amd64 Packages [5.816 B]     
Get:8 http://security.ubuntu.com/ubuntu artful-security/universe Translation-en [4.412 B]
Get:9 http://security.ubuntu.com/ubuntu artful-security/universe amd64 DEP-11 Metadata [2.184 B]
Get:10 http://security.ubuntu.com/ubuntu artful-security/universe DEP-11 64x64 Icons [4.058 B]
Get:11 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 Packages [8.103 kB]                   
Get:12 http://de.archive.ubuntu.com/ubuntu artful/universe Translation-en [4.789 kB]                   
Get:13 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 DEP-11 Metadata [2.845 kB]            
Get:14 http://de.archive.ubuntu.com/ubuntu artful/universe DEP-11 64x64 Icons [7.915 kB]               
Get:15 http://de.archive.ubuntu.com/ubuntu artful-updates/universe i386 Packages [12,6 kB]             
Get:16 http://de.archive.ubuntu.com/ubuntu artful-updates/universe amd64 Packages [12,6 kB]            
Get:17 http://de.archive.ubuntu.com/ubuntu artful-updates/universe Translation-en [8.036 B]            
Get:18 http://de.archive.ubuntu.com/ubuntu artful-updates/universe amd64 DEP-11 Metadata [5.088 B]     
Get:19 http://de.archive.ubuntu.com/ubuntu artful-updates/universe DEP-11 64x64 Icons [7.399 B]        
Fetched 31,9 MB in 33s (955 kB/s)                                                                      
Reading package lists... Done
alexander@qemu:~$ sudo apt-get install flatpak
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  bubblewrap libostree-1-1 xdg-desktop-portal xdg-desktop-portal-gtk
The following NEW packages will be installed:
  bubblewrap flatpak libostree-1-1 xdg-desktop-portal xdg-desktop-portal-gtk
0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
Need to get 942 kB of archives.
After this operation, 4.520 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 bubblewrap amd64 0.2.0-1 [32,9 kB]
Get:2 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 libostree-1-1 amd64 2017.12-1ubuntu1 [214 kB]
Get:3 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 flatpak amd64 0.8.7-5 [534 kB]
Get:4 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 xdg-desktop-portal amd64 0.8-2 [83,3 kB]
Get:5 http://de.archive.ubuntu.com/ubuntu artful/universe amd64 xdg-desktop-portal-gtk amd64 0.7-2 [77,6 kB]
Fetched 942 kB in 1s (632 kB/s)              
Selecting previously unselected package bubblewrap.
(Reading database ... 126008 files and directories currently installed.)
Preparing to unpack .../bubblewrap_0.2.0-1_amd64.deb ...
Unpacking bubblewrap (0.2.0-1) ...
Selecting previously unselected package libostree-1-1:amd64.
Preparing to unpack .../libostree-1-1_2017.12-1ubuntu1_amd64.deb ...
Unpacking libostree-1-1:amd64 (2017.12-1ubuntu1) ...
Selecting previously unselected package flatpak.
Preparing to unpack .../flatpak_0.8.7-5_amd64.deb ...
Unpacking flatpak (0.8.7-5) ...
Selecting previously unselected package xdg-desktop-portal.
Preparing to unpack .../xdg-desktop-portal_0.8-2_amd64.deb ...
Unpacking xdg-desktop-portal (0.8-2) ...
Selecting previously unselected package xdg-desktop-portal-gtk.
Preparing to unpack .../xdg-desktop-portal-gtk_0.7-2_amd64.deb ...
Unpacking xdg-desktop-portal-gtk (0.7-2) ...
Setting up libostree-1-1:amd64 (2017.12-1ubuntu1) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Processing triggers for desktop-file-utils (0.23-1ubuntu3) ...
Setting up bubblewrap (0.2.0-1) ...
Processing triggers for libc-bin (2.26-0ubuntu2) ...
Processing triggers for doc-base (0.10.7) ...
Processing 1 added doc-base file...
Processing triggers for man-db (2.7.6.1-2) ...
Processing triggers for gnome-menus (3.13.3-6ubuntu5) ...
Processing triggers for dbus (1.10.22-1ubuntu1) ...
Setting up flatpak (0.8.7-5) ...
flatpak-system-helper.service is a disabled or a static unit, not starting it.
Setting up xdg-desktop-portal (0.8-2) ...
Setting up xdg-desktop-portal-gtk (0.7-2) ...
Processing triggers for dbus (1.10.22-1ubuntu1) ...
alexander@qemu:~$ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
alexander@qemu:~$ flatpak install flathub com.valvesoftware.Steam
Required runtime for com.valvesoftware.Steam/x86_64/stable (org.freedesktop.Platform/x86_64/1.6) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]: y
Installing: org.freedesktop.Platform/x86_64/1.6 from flathub

8 delta parts, 63 loose fetched; 130057 KiB transferred in 189 seconds                                  
Installing: org.freedesktop.Platform.VAAPI.Intel/x86_64/1.6 from flathub

1 delta parts, 3 loose fetched; 2337 KiB transferred in 35 seconds                                      
Installing: org.freedesktop.Platform.Locale/x86_64/1.6 from flathub

5 metadata, 1 content objects fetched; 14 KiB transferred in 2 seconds                                  
Installing: com.valvesoftware.Steam/x86_64/stable from flathub

1 delta parts, 3 loose fetched; 2714 KiB transferred in 7 seconds                                       
Installing: org.freedesktop.Platform.Compat32/x86_64/1.6 from flathub

8 delta parts, 56 loose fetched; 126146 KiB transferred in 128 seconds                                  
alexander@qemu:~$ flatpak run com.valvesoftware.Steam 
Setting up Steam content in /home/alexander/.var/app/com.valvesoftware.Steam/data/Steam`

**Comment by charlesjt** [comment]
@probonopd I don't think you should expect much from Canonical, it is known for promoting its technologies without even looking at the rest, but there is a PPA and Universe repo that always stays updated to be able to use Flatpak. Even in fedora they exist in the official repositories for snap/snapd even though Red Hat's bet is towards Flatpak. It is one more reason not to opt for Snap, considering that derivatives such as Linux Mint will opt for flatpak or others like Pop_OS if they will provide support.

**Comment by probonopd** [comment]
@Alexander-Wilms sure it can probably be made to work but it is not an out-of-the-box experience that "just works", at least of today. @charlesjt I am not expecting anything from distributions that would make "sideloading" easier, since that is unlikely to be a distribution's prevailing mindset.

**Comment by AsciiWolf** [comment]
@Alexander-Wilms Flatpak package from the universe repository is, unfortunately, outdated. :-(

**Comment by charlesjt** [comment]
@probonopd Then I dont understand why you take the trouble to see if something works out of the box or not. Precisely my comment suggests that Canonical wants Snap to work out of the box... Flatpak no.

**Comment by silvernode** [comment]
I use Flatpak, Snapd, and Appimage for maximum program availability. The only case when I can't use snapd is on any system that isn't already listed. For example right now I am running Void Linux which does not have snapd but does have flatpak. I would use snapd if it was available, and that's the biggest problem with snaps right now. Flatpak seems to be in a lot more repositories thn snapd but I think Valve should make a snap and a flatpak anyway and possibly kill the deb file (to make their life easier). This way they can support *Linux instead of just Debian. Include all the dependencies and runtimes they need in their steam container to ensure everything works as smooth as possible. Sure it might be a very large container but at least you'd have a better chance of things working the first time.

**Comment by jakobjakobson13** [comment]
+1 for flatpak

**Comment by Hasshu** [comment]
An anecdote: the other day, one of the RPM Fusion¹ maintainers responsible for the Steam package had a galaxy-brain moment, and decided to make `gnome-shell-extension-gamemode` a recommended² package. Which means that by default their `steam` package would pull in an [irrelevant shell extension](/gicmo/gamemode-extension) from the repo... and the [entire GNOME 3 desktop](https://www.reddit.com/r/Fedora/comments/gptjd6/why_the_update_want_to_install_a_lot_of_gnome/) together with itself, if you happen to use a different DE. Moreover, a [bug](https://bugzilla.rpmfusion.org/show_bug.cgi?id=5659) was filed only 5 days later, and it took them another 6 days to move the fixed package to the stable repo. _11 days to [type a few words](https://github.com/rpmfusion/steam/commit/92ff32341eba3b8e8dca9d63ea622e3a99a58643) and undo this blunder._

Something like that would never happen, if there were an _official **distro-agnostic**_ way of installing Steam. Having to depend on whims of a third party, that may — or may not — package a piece of software properly and keep it up-to-date, is an ugly and ridiculous anachronism.

¹ [RPM Fusion](https://rpmfusion.org/RPM%20Fusion) is a semi-official Fedora repository, and the _de facto_ source for not-quite-free software.
² [Until recently](/rpmfusion/steam/pull/11), their `steam` package used to _require_ `gamemode`. Go figure.

**Comment by jjardon** [comment]
@Hasshu if it helps, please remember an unofficial flatpak package already exists: https://github.com/ValveSoftware/steam-for-linux/issues/4473#issuecomment-309249157

**Comment by Hasshu** [comment]
@jjardon Thanks, I think I've heard about this one, but what interests me is _an **official** distro-agnostic way to install Steam_. It's just so happens that Flatpak might fit the bill.

**Comment by jjardon** [comment]
@Hasshu understood, maybe this can be if your interest: https://github.com/ValveSoftware/steam-runtime/blob/master/doc/possible-designs.md

**Comment by Hasshu** [comment]
@jjardon So the developers are considering something like that? Sounds promising.

**Comment by xioren** [comment]
2023 and I'd like to show continued support for this idea. I think it benefits both users and Valve. Containerization is the future and many apps I use have already made the jump to Flatpak.

Distributing Steam as an official Flatpak would consolidate the process of both packaging, distributing and maintaining Steam for Linux as well as installation. Some distros already maintain their own Steam package (Debian for example) and many if not most users who wish could still install Steam via their respective package manager. In terms of SteamOS, I don't believe this would conflict as it would make SteamOS similar to Fedora [Silverblue](https://silverblue.fedoraproject.org/); an immutable system with Flatpak packaging on top.

**Comment by jakobjakobson13** [comment]
@kisak-valve : Are there any news on this topic?

**Comment by ryanabx** [comment]
The unofficial flatpak works quite well, but I'm hoping that steam will embrace the flatpak as official one of these days.

**Comment by perroboc** [comment]
Given the latest news[^1] that the Snap implementation sucks, please reconsider officialy distributing Steam through Flatpak 🙏 

[^1]: [Valve seeing increasing bug reports due to Steam Snap - other methods recommended](https://www.gamingonlinux.com/2024/01/valve-seeing-increasing-bug-reports-due-to-steam-snap-other-methods-recommended/)

**Comment by AsciiWolf** [comment]
@perroboc That would not solve the problem for Ubuntu users since Flatpak is unsupported on Ubuntu and even though its support can be added, it is far from being ideal.

**Comment by xioren** [comment]
@AsciiWolf I don't agree with that framing. It's not that Flatpak is "unsupported" on Ubuntu, it just doesn't come installed by default. Just: 
```
apt install flatpak && flatpak install com.valve.Steam
```
and you're done. Hardly "far from being ideal" and in fact still easier to install and manage than any current approach.

**Comment by AsciiWolf** [comment]
No, it's really not that easy and there are many more problems with Flatpak support on Ubuntu. For example, no support for managing Flatpak software using GUI (Ubuntu Software / App Center). Also, the Flatpak package itself is in universe and as such tends to get really outdated (including unfixed CVEs) and problematic (along with xdg desktop portals, but at least their support will hopefully improve since Snap also started using them). It is best to avoid Flatpak completely on Ubuntu (and avoid Snap on other distributions except Ubuntu).

**Comment by eternal-sorrow** [comment]
>It is best to avoid Flatpak completely on Ubuntu (and avoid Snap).

FTFY

**Comment by xioren** [comment]
@AsciiWolf Well if you feel if it is necessary for Steam to be installable via Ubuntus GUI package manager, then that is something Ubuntu users need to make clear to Canonical. Ultimately Valve will package Steam in the way that they deem appropriate. Different distros can choose to provide Steam (or not), either by maintaining their own repackaging of Steam (Snap, Apt, etc) or providing Steams official package. So your complaints seem to be with how Ubuntu makes Steam available on their distribution and not how Valve is packaging Steam.

**Comment by AsciiWolf** [comment]
@xioren My complains are about your first comment suggesting Flatpak as a possible solution for Ubuntu users. I agree that this should ideally be discussed with Canonical, however I doubt they will ever consider supporting Flatpak. Anyway, let's stop this. It is offtopic here and also a pointless discussion.
